tools/c7n_gcp/tests/test_resource.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gcp_common import BaseTest
+from c7n_gcp.provider import resources
+
+
+class ResourceMetaTest(BaseTest):
+
+    def test_resource_id_meta(self):
+        missing = []
+        for name, resource in resources.items():
+            if not getattr(resource.resource_type, 'id', None):
```

Comment:
Hello @kapilt,
There is a case when an API response does not have an exact identifier.
As an example [GCP -> Stackdriver -> logs.list](https://cloud.google.com/logging/docs/reference/v2/rest/v2/logs/list) returns ListLogsResponse. ListLogsResponse contains only one property logNames which is array and can't be used as a identifier.
Maybe, in this case, we can set id as None but tests will failure. Could you please change the checking of attribute value on attribute existence.

Reply:
strings are not a resource, convert to dictionary and use the name as the id.